nntrainer/tensor/manager.h

```diff
@@ -223,6 +223,41 @@ class Manager {
   std::vector<Weight *> getWeights();
 
   /**
+   * @brief Get the Min Max of a tensor execution order
+   *
+   * @param name name of the
+   * @return std::pair<unsigned int, unsigned int>
+   */
+  std::pair<unsigned int, unsigned int>
+  getMinMaxTensorExecutionOrder(const std::string &name);
+
+  /**
+   * @brief check if given execution order is the first access
+   *
+   * @param name tensor name
+   * @param current_execution current execution
+   * @return bool true if given execution order first access
+   */
+  bool isFirstAccess(const std::string &name, unsigned current_execution) {
+    /** NYI!! */
+    /// @todo add cache machanism
```

Comment:
Instead of caching, consider sorting the execution orders in tensor_pool. 
This would also reduce getting min and max to constant time operations.

Reply:
yes, we can sort this. I'll update todo